src/main/java/com/microfocus/octane/plugins/admin/ConfigResource.java

```diff
@@ -120,27 +122,52 @@ public Response getWorkspaceConfigurationById(@PathParam("id") String id) {
 
     @GET
     @Path("/workspaces/supported-octane-types")
-    public Response getSupportedOctaneTypes(@QueryParam("space-config-id") String spaceConfigurationId, @QueryParam("workspace-id") long workspaceId, @QueryParam("udf-name") String udfName) {
+    public Response getSupportedOctaneTypes(@QueryParam("space-config-id") String spaceConfigurationId, @QueryParam("workspace-ids") String workspaceIds, @QueryParam("udf-name") String udfName) {
         if (!hasPermission()) {
             return Response.status(Status.FORBIDDEN).build();
         }
 
+        List<Long> workspaceIdsList;
+        try {
+            workspaceIdsList = Arrays.stream(workspaceIds.split(",")).map(Long::parseLong).collect(Collectors.toList());
+        } catch (NumberFormatException e) {
+            log.error(e.getMessage());
+
+            return Response.serverError().entity("Workspace id not a valid number: " + e.getMessage()).build();
+        }
+
         SpaceConfiguration sc = ConfigurationManager.getInstance().getSpaceConfigurationById(spaceConfigurationId, true).get();
-        List<String> types = OctaneRestManager.getSupportedOctaneTypes(sc, workspaceId, udfName);
-        List<String> names = types.stream().map(t -> OctaneEntityTypeManager.getByTypeName(t).getLabel()).sorted().collect(Collectors.toList());
+        List<Set<String>> types = workspaceIdsList.stream().map(wsId -> OctaneRestManager.getSupportedOctaneTypes(sc, wsId, udfName)).collect(Collectors.toList());
+
+        Set<String> commonEntityTypes = ConfigurationUtil.retainAllSets(types);
+
+        List<String> names = commonEntityTypes.stream().map(t -> OctaneEntityTypeManager.getByTypeName(t).getLabel()).sorted().collect(Collectors.toList());
+
         return Response.ok(names).build();
     }
 
     @GET
     @Path("/workspaces/possible-jira-fields")
-    public Response getPossibleJiraFields(@QueryParam("space-config-id") String spaceConfigurationId, @QueryParam("workspace-id") long workspaceId) {
+    public Response getPossibleJiraFields(@QueryParam("space-config-id") String spaceConfigurationId, @QueryParam("workspace-ids") String workspaceIds) {
         if (!hasPermission()) {
             return Response.status(Status.FORBIDDEN).build();
         }
 
+        List<Long> workspaceIdsList;
+        try {
+            workspaceIdsList = Arrays.stream(workspaceIds.split(",")).map(Long::parseLong).collect(Collectors.toList());
```

Comment:
what happens if workspaceIds comes as null or invalid value?

Reply:
it can't be null object, only empty string
and if it's anything other than numbers it will throw an error: 
`Workspace id not a valid number: For input string: "sdadas"`